src/core_functions/aggregate/README.md

```diff
@@ -0,0 +1,221 @@
+* As the only elements of a query _without_ a `GROUP BY` clause (simple aggregation)
+* Modified by an `OVER` clause (windowed aggregation)
+* As an argument to the `list_aggregate` function (list aggregation)
+
+## Aggregation Operations
+
+In order to define an aggregate function, you need to define some operations.
+These operations accumulate data into a _state_ object that is specific to the aggregate.
+Unlike simple scalar functions, there are several of these:
+
+| Operation | Description | Required | 
+| :-------- | :---------- | :------- | 
+| `size`  | Returns the fixed size of the `State` | X | 
+| `initialize` | Constructs the `State` in raw memory | X |
+| `destructor` | Destructs the `State` back to raw memory |  |
+| `update` | Accumulate the arguments into the corresponding `State` | X |
+| `simple_update` | Accumulate the arguments into a single `State`. |  |
+| `combine` | Merge one `State` into another |  |
+| `finalize` | Convert a `State` into a final value. | X |
+| `window` | Compute a windowed aggregate value from the inputs and frame bounds |  |
+| `bind` | Modify the binding of the aggregate |  |
+| `statistics` | Derive statistics of the result from the statistics of the arguments |  |
+| `serialize` | Write a `State` to a relocatable binary blob |  |
+| `deserialize` | Read a `State` from a binary blob |  |
+
+In addition to these high level functions,
+there is also a template `AggregateExecutor` that can be used to generate these functions
+from row-oriented static methods in a class.
+
+There are also a number of helper objects that contain various bits of context for the aggregate,
+such as binding data and extracted validity masks.
+By combining them into these helper objects, we reduce the number of arguments to various functions.
+The helpers can vary by the number of arguments, and we will refer to them simply as `info` below.
+Consult the code for details on what is available.
+
+### Size
+
+```cpp
+size()
+```
+
+`State`s are allocated in memory blocks by the various operators
+so each aggregate has to tell the operator how much memory it will require.
+Note that this is just the memory that the aggregate needs to get started -
+it is perfectly legal to allocate variable amounts of memory
+and storing pointers to it in the `State`.
+
+### Initialize
+
+```cpp
+initialize(State *)
+```
+
+Construct a _single_ empty `State` from uninitialized memory.
+
+### Destructor
+
+```cpp
+destructor(Vector &state, AggregateInputData &info, idx_t count)
+```
+
+Destruct a `Vector` of state pointers.
+If you are using a template, the method has the signature
+
+```cpp
+Destroy(State &state, AggregateInputData &info)
+```
+
+### Update and Simple Update
+
+```cpp
+update(Vector inputs[], AggregateInputData &info, idx_t ninputs, Vector &states, idx_t count)
+```
+
+Accumulate the input arguments for each row into the corresponding `State`.
```

Comment:
Can we add something along these lines: ', which can be found at the corresponding index in the `states` vector'. Just to make it more clear that the input to this function already maps the states to the input rows?